src/libstd/internal_rand.rs

```diff
@@ -0,0 +1,275 @@
+// Copyright 2013-2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Interface to the operating system provided random number
+//! generators:
+//!
+//! - Unix-like systems (Linux, Android, Mac OSX): read directly from
+//!   `/dev/urandom`, or from `getrandom(2)` system call if available.
+//! - Windows: calls `CryptGenRandom`, using the default cryptographic
+//!   service provider with the `PROV_RSA_FULL` type.
+//! - iOS: calls SecRandomCopyBytes as /dev/(u)random is sandboxed.
+//! - OpenBSD: uses the `getentropy(2)` system call.
+
+
+pub use self::imp::fill_bytes;
+
+#[cfg(all(unix, not(target_os = "ios"), not(target_os = "openbsd")))]
```

Comment:
As I mention above, I was/am going to remove the existing support from std since we only need what's in this file now. However, the `rand` crate will still need to exist for internal testing purposes; should it still be dedup'd with that? (It's possible, it just means slightly more boilerplate in here.)